File: java/SelectionSort.java
class SelectionSort{
	public static void printArray(int a[]){
		System.out.println("Array: ");
		for(int i=0; i<a.length; i++){
			System.out.print(a[i]+" ");
		}
			System.out.println("\n");		
	}

	public static void sort(int[] arr){
		int size=arr.length;
		int temp=0;

		for(int i=0; i<size-1; i++){
			int mindex=i;
			for(int x=i+1; x<size; x++){
				if(arr[x]<arr[mindex])
					mindex=x;				
			}

			temp=arr[mindex];
			arr[mindex]=arr[i];
			arr[i]=temp;

		}

	}

	public static void main(String args[]){
		int[] arr={28,41,53,4,99,15};
		printArray(arr);
		sort(arr);
		printArray(arr);
	}
}

File: cpp/selectionSort.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){

    int n; 
    cout<<"Enter size\n ";
    cin>>n;

    int arr[n];
    cout<<"Enter array elements \n ";
    for (int i = 0; i < n; i++)
    {
    cin>>arr[i];
    }

    //sort

    for(int i=0;i<n;i++)
    {
    for(int j=i+1;j<n;j++)
    {
        if(arr[j]<arr[i])
        {
            int temp=arr[j];
            arr[j]=arr[i];
            arr[i]=temp;
            
        }


    }

    }

    cout<<"SORTED array elements \n ";
    for (int i = 0; i < n; i++)
    {
    cout<<arr[i]<<" ";
    }

return 0;
}